sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala

```diff
@@ -127,24 +171,41 @@ class RapidsShuffleIterator(
             * about it.
             */
           blockId._1 match {
-            case sbbid: ShuffleBlockBatchId => sbbid
+            case sbbid: ShuffleBlockBatchId => (sbbid, blockId._3)
```

Comment:
Nit: We should consider using a case class.  I find generic tuples beyond 2 to often be hard to follow without useful names for things, e.g.: needing to recall what the `Int` field really means.  A `case class` makes that obvious when reading the code.  Tuples not so much.  Doesn't have to be done as part of this PR.

Reply:
Sounds good.